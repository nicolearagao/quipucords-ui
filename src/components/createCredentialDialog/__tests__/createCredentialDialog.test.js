import React from 'react';
import { CreateCredentialDialog, authenticationTypeOptions, becomeMethodTypeOptions } from '../createCredentialDialog';

describe('CreateCredentialDialog Component', () => {
  it('should render a basic component', async () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'network'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect(component.screen.render()).toMatchSnapshot('basic');
  });

  it('should export select options', () => {
    expect({
      authenticationTypeOptions,
      becomeMethodTypeOptions
    }).toMatchSnapshot('options');
  });

  it('should handle a pending display state', () => {
    const props = {
      useCredential: () => ({
        pending: true,
        show: true,
        add: true,
        credentialType: 'satellite'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect(component.screen.render().querySelector('.pf-c-modal-box__body')).toMatchSnapshot('pending');
  });

  it('should handle variations in basic form display for network credential type', () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'network'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect([
      ...component.screen.render().querySelectorAll('input'),
      ...component.screen.render().querySelectorAll('.quipucords-select')
    ]).toMatchSnapshot('network');
  });

  it('should handle variations in basic form display for openshift credential type', () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'openshift'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect([
      ...component.screen.render().querySelectorAll('input'),
      ...component.screen.render().querySelectorAll('.quipucords-select')
    ]).toMatchSnapshot('openshift');
  });

  it('should handle variations in basic form display for satellite credential type', () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'satellite'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect([
      ...component.screen.render().querySelectorAll('input'),
      ...component.screen.render().querySelectorAll('.quipucords-select')
    ]).toMatchSnapshot('satellite');
  });

  it('should handle variations in basic form display for vcenter credential type', () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'vcenter'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect([
      ...component.screen.render().querySelectorAll('input'),
      ...component.screen.render().querySelectorAll('.quipucords-select')
    ]).toMatchSnapshot('vcenter');
  });

  it('should handle variations in basic form display for ansible credential type', () => {
    const props = {
      useCredential: () => ({
        show: true,
        add: true,
        credentialType: 'ansible'
      })
    };

    const component = renderComponent(<CreateCredentialDialog {...props} />);
    expect([
      ...component.screen.render().querySelectorAll('input'),
      ...component.screen.render().querySelectorAll('.quipucords-select')
    ]).toMatchSnapshot('ansible');
  });
});
