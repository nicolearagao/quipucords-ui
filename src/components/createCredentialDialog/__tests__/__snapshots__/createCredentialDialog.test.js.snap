// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCredentialDialog Component should export select options: options 1`] = `
{
  "authenticationTypeOptions": [
    {
      "title": [Function],
      "value": "sshKey",
    },
    {
      "title": [Function],
      "value": "token",
    },
    {
      "selected": true,
      "title": [Function],
      "value": "usernamePassword",
    },
  ],
  "becomeMethodTypeOptions": [
    "sudo",
    "su",
    "pbrun",
    "pfexec",
    "doas",
    "dzdo",
    "ksu",
    "runas",
  ],
}
`;

exports[`CreateCredentialDialog Component should handle a pending display state: pending 1`] = `
<div
  class="pf-c-modal-box__body"
  id="pf-modal-part-3"
>
  <div>
    <form
      class="pf-c-form pf-m-horizontal"
      novalidate=""
    >
      <div
        class="pf-c-empty-state quipucords-empty-state"
      >
        <div
          class="pf-c-empty-state__content"
        >
          <span
            aria-hidden="true"
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-xl pf-c-empty-state__icon"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            >
              <span
                class="pf-c-spinner__lead-ball"
              >
                <span
                  class="pf-c-spinner__tail-ball"
                />
                <h3
                  class="pf-c-title pf-m-lg"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                   t(form-dialog.empty-state_title, {"context":"pending"}) 
                </h3>
              </span>
            </span>
          </span>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`CreateCredentialDialog Component should handle variations in basic form display for ansible credential type: ansible 1`] = `
[
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"ansible\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-24"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value="t(form-dialog.label, {\\"
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-25"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="cred_name"
    maxlength="64"
    name="name"
    placeholder="t(form-dialog.label_name_create-credential, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-26"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="username"
    name="username"
    placeholder="t(form-dialog.label_username, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-27"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="password"
    name="password"
    placeholder="t(form-dialog.label_password, {\\"
    type="password"
    value=""
  />,
]
`;

exports[`CreateCredentialDialog Component should handle variations in basic form display for network credential type: network 1`] = `
[
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"network\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-7"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value="t(form-dialog.label, {\\"
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-8"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="cred_name"
    maxlength="64"
    name="name"
    placeholder="t(form-dialog.label_name_create-credential, {\\"
    type="text"
    value=""
  />,
  <div
    class="quipucords-select quipucords-select__not-inline"
  >
    <div
      class="pf-c-select quipucords-select-pf quipucords-select-pf__position-down "
      data-ouia-component-id="OUIA-Generated-Select-single-3"
      data-ouia-component-type="PF4/Select"
      data-ouia-safe="true"
    />
  </div>,
]
`;

exports[`CreateCredentialDialog Component should handle variations in basic form display for openshift credential type: openshift 1`] = `
[
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"openshift\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-13"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value="t(form-dialog.label, {\\"
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-14"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="cred_name"
    maxlength="64"
    name="name"
    placeholder="t(form-dialog.label_name_create-credential, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-15"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="auth_token"
    name="auth_token"
    placeholder="t(form-dialog.label_token_create-credential, {\\"
    type="text"
    value=""
  />,
]
`;

exports[`CreateCredentialDialog Component should handle variations in basic form display for satellite credential type: satellite 1`] = `
[
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"satellite\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-16"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value="t(form-dialog.label, {\\"
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-17"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="cred_name"
    maxlength="64"
    name="name"
    placeholder="t(form-dialog.label_name_create-credential, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-18"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="username"
    name="username"
    placeholder="t(form-dialog.label_username, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-19"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="password"
    name="password"
    placeholder="t(form-dialog.label_password, {\\"
    type="password"
    value=""
  />,
]
`;

exports[`CreateCredentialDialog Component should handle variations in basic form display for vcenter credential type: vcenter 1`] = `
[
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"vcenter\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-20"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value="t(form-dialog.label, {\\"
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-21"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="cred_name"
    maxlength="64"
    name="name"
    placeholder="t(form-dialog.label_name_create-credential, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-22"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="username"
    name="username"
    placeholder="t(form-dialog.label_username, {\\"
    type="text"
    value=""
  />,
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-23"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="password"
    name="password"
    placeholder="t(form-dialog.label_password, {\\"
    type="password"
    value=""
  />,
]
`;

exports[`CreateCredentialDialog Component should render a basic component: basic 1`] = `
<screen>
  <div>
    <div
      class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
    >
      <div>
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-modal-part-2"
              aria-label="t(modal.aria-label-default)"
              aria-labelledby="pf-modal-part-0"
              aria-modal="true"
              class="pf-c-modal-box pf-m-align-top"
              data-ouia-component-id="OUIA-Generated-Modal-1"
              data-ouia-component-type="PF4/ModalContent"
              data-ouia-safe="true"
              id="pf-modal-part-0"
              role="dialog"
              style="--pf-c-modal-box--m-align-top--spacer: 5%;"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="OUIA-Generated-Modal-1-ModalBoxCloseButton"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  />
                </svg>
              </button>
              <header
                class="pf-c-modal-box__header"
              >
                <h4
                  class="pf-c-title pf-m-md"
                  data-ouia-component-id="OUIA-Generated-Title-1"
                  data-ouia-component-type="PF4/Title"
                  data-ouia-safe="true"
                >
                   t(form-dialog.title, {"context":"create-credential"}) 
                </h4>
              </header>
              <div
                class="pf-c-modal-box__body"
                id="pf-modal-part-2"
              >
                <div>
                  <form
                    class="pf-c-form pf-m-horizontal"
                    novalidate=""
                  >
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="generatedid-"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                             t(form-dialog.label_source-type, {"context":"create-credential"}) 
                          </span>
                        </label>
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <input
                          aria-invalid="false"
                          class="pf-c-form-control quipucords-form__text-input "
                          context\\":\\"network\\"})"=""
                          data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe="true"
                          id="generatedid-"
                          name="generatedid-"
                          readonly=""
                          type="text"
                          value="t(form-dialog.label, {\\"
                        />
                      </div>
                    </div>
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="cred_name"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                             t(form-dialog.label_name, {"context":"create-credential"}) 
                          </span>
                        </label>
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <input
                          aria-invalid="false"
                          class="pf-c-form-control quipucords-form__text-input "
                          context\\":\\"placeholder\\"})"=""
                          data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                          data-ouia-component-type="PF4/TextInput"
                          data-ouia-safe="true"
                          id="cred_name"
                          maxlength="64"
                          name="name"
                          placeholder="t(form-dialog.label_name_create-credential, {\\"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="generatedid-"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                             t(form-dialog.label_auth-type, {"context":"create-credential"}) 
                          </span>
                        </label>
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <div
                          class="quipucords-select quipucords-select__not-inline"
                        >
                          <div
                            class="pf-c-select quipucords-select-pf quipucords-select-pf__position-down "
                            data-ouia-component-id="OUIA-Generated-Select-single-1"
                            data-ouia-component-type="PF4/Select"
                            data-ouia-safe="true"
                          />
                        </div>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;
